### PR TITLE
session: admin check table when modify variables

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

add `admin check table ` for each set global sys vars

### What is changed and how it works?

add `admin check table` for each set global sys vars

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - N/A

Side effects

 - N/A

Related changes

 - Need to cherry-pick to the release branch

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/9589)

### CODE REVIEW

File: session/bootstrap.go [hunk 1]
```diff
@@ -788,6 +789,14 @@ func doDDLWorks(s Session) {
 // doDMLWorks executes DML statements in bootstrap stage.
 // All the statements run in a single transaction.
 func doDMLWorks(s Session) {
+	if config.CheckTableBeforeDrop {
+		defer func() {
+			sql := fmt.Sprintf("admin check table `%s`.`%s`", mysql.SystemDB, mysql.GlobalVariablesTable)
+			log.Warnf("%s before AfterInit GlobalSysVar", sql)
```
Comment (zz-jason) [inline session/bootstrap.go]: should we change the log level to INFO?

File: session/session.go [hunk 2]
```diff
@@ -848,6 +848,14 @@ func (s *session) SetGlobalSysVar(name, value string) error {
 	sql := fmt.Sprintf(`REPLACE %s.%s VALUES ('%s', '%s');`,
 		mysql.SystemDB, mysql.GlobalVariablesTable, name, sVal)
 	_, _, err = s.ExecRestrictedSQL(s, sql)
+	if err != nil && config.CheckTableBeforeDrop {
```
Comment (winkyao) [inline session/session.go]: Why only admin check table when err != nil?

**Comment by lysu** [comment]
this PR meet some gofail question and need be address later, pending a mount..- -

**Comment by jackysp** [comment]
may be related to https://github.com/pingcap/tidb/pull/10278

**Pull request closed without merging**